### DISCUSSION TITLE
Lesson 4: The transaction declared chain ID 5777, but the connected node is on 1337

### DISCUSSION THREAD

**Discussion by GODZAOZAO** [smartcontractkit/full-blockchain-solidity-course-py#1127] [Q&A] [answered]
Hi, all,
I'm getting this error "The transaction declared chain ID 5777, but the connected node is on 1337". I've been checking my Ganache and changed my chain_id to 5777 and HTTPProvider to http://127.0.0.1:7545. Should I change my chain_id back to 1337?

I also don't know why the connected node is still on 1337 since I have changed the "my_address" and "private_key" values according to the first address and private key shown in my Ganache.

<img width="1728" alt="Screen Shot 2022-02-21 at 2 01 12 PM" src="https://user-images.githubusercontent.com/45237568/155013372-48ebcb48-6881-4778-9c60-9ad575a78029.png">
<img width="1728" alt="Screen Shot 2022-02-21 at 2 01 24 PM" src="https://user-images.githubusercontent.com/45237568/155013374-10b91818-da82-4961-a205-376dd2557009.png">

**Comment by cromewar** [comment, marked as answer]
Is better to change the ganache configuration to:
ChainID: 1337
and use port: 8545

ALso based on your error you are facing the cytools error, is a detailed guide about how to fix it here: 
https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#python-related-issues

Cheers.

**Comment by JohnEMagee** [comment]
@GODZAOZAO I wonder are you using a Mac M1, because I am and I got the exact same answers.  If this is true, if someone could do this course in an M1 and provide all the alternatives that would be awesome.

If you manually change the chain_id = 1337 and include the gasPrice variable you do not have to change the HTTPProvider.  But if someone could provide an explanation for WHY this doesn't work that would be great.  I had to finally install rosetta to use ganache but if it's not working properly that would be disappointing.

**Reply by cromewar** [reply]
Hard to say, the architecture on M1 Macs is strange and really new so they will have some weird problems Apple should solve.
Regarding to the chain ID change, is just for standardization, brownie uses 1337 by default and will not use other one unless we explicitly say so.
